### PR TITLE
Oauth code tidy up

### PR DESCRIPTION
**Pull request by zas**
# Summary

* This is a…
  * [ ] Bug fix
  * [ ] Feature addition
  * [x] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-XXX

# Solution

# Action

### CODE REVIEW

File: picard/oauth.py [hunk 1]
```diff
@@ -172,28 +177,37 @@ def set_access_token(self, access_token, expires_in):
...
+        log.debug("OAuth: refreshing access_token with a refresh_token %s", self.refresh_token)
+        path = "/oauth2/token"
+        params = {
+            "grant_type": "refresh_token",
+            "refresh_token": self.refresh_token,
+            "client_id": MUSICBRAINZ_OAUTH_CLIENT_ID,
+            "client_secret": MUSICBRAINZ_OAUTH_CLIENT_SECRET,
+        }
+        self.webservice.post(
+            self.host, self.port, path, self._query_data(params),
```
Comment (phw) [inline picard/oauth.py:199]: Instead of `self._query_data` we could also use `urllib.parse.urlencode` and get rid of the custom Qt based implementation.

**Review by phw — CHANGES_REQUESTED** [review]

File: picard/oauth.py (hunk 1 above)
Comment (zas) [reply picard/oauth.py:190]: Yup, agreed.

See d1de2a7208 and 8b787482f

File: test/test_oauth.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+from picard.oauth import OAuthManager
+
+
+class OAuthManagerTest(PicardTestCase):
+
+    def test_query_data(self):
+        params = {
+            'a&b': 'a b',
+            'c d': 'c&d',
+            '': '',
```
Comment (phw) [inline test/test_oauth.py:33]: I'd add some test entries which include `=` here, to make sure that gets encoded as well

**Review by phw — APPROVED** [review]

File: test/test_oauth.py (hunk 2 above)
Comment (zas) [reply test/test_oauth.py:34]: Done in 53ec0e0

**Review by phw — APPROVED** [review]

**Pull request merged**